README.md

```diff
@@ -0,0 +1,189 @@
+# Ballot
+
+## Overview
+
+`ballot` ensures that it's the only one running a command, by running a leader
+election on a ZooKeeper cluster. Others that try to run the same command (based
+on ZooKeeper path) will wait until the active `ballot` releases leadership.
+
+Some example use cases include:
+
+- A distributed cron service, where jobs need to run at specific times or time
+  intervals, and cannot overlap, but still needs to failover to standby
+  instances on failure. In this case `ballot` acts as
+  [lockrun](http://www.unixwiz.net/tools/lockrun.html) would on a single-server
+  system.
+- Any singleton service, that would be a kubernetes `Deployment` of 1 replica,
+  if kubernetes is not an option, or if running on kubernetes and trying to
+  achieve failover faster than the control plane can re-schedule pods.
+
+### Examples
+
+```shell
+ballot run once --candidate-id `hostname` -- /usr/bin/env LD_PRELOAD=trickle.so command-with-exclusive-resource
+ballot run cron --candidate-id `hostname` --schedule='@daily' -- /usr/bin/env LD_PRELOAD=trickle.so command-with-exclusive-resource
+```
+
+## Pre-requisites
+
+A running and healthy [ZooKeeper](https://zookeeper.apache.org/) ensemble is
+required. Only plain unauthenticated connections are supported at this time.
+
+## Configuration
+
+`ballot` relies on
+[viper](https://github.com/spf13/viper)/[cobra](https://github.com/spf13/cobra)
+for command line and configuration, and thus can be configured using command
+line arguments, environment variables, and configuration files. Remote
+configuration sources are not available at this time.
+
+Several configuration sources can be mixed, as long as parameters are not
+duplicated between sources (in which case resolution order is subject to change
```

Comment:
This could be due to the deps used, but changing the order config is loaded `without notice` seems like a bad choice. If the intent is to make clear that this *could* change in the future and to keep up with docs/release notes when upgrading, I think this could be made clearer.

Reply:
It's more that there's no test coverage on this particular aspect, which makes it difficult to guarantee an order. But it's definitely planned to have the order predictable. I'll rephrase.

Reply:
I just removed the sentence since we're approaching 1.0.